cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/profile/ProfileDataset.java

```diff
@@ -123,10 +123,15 @@ public Profile getProfile(ProfileId profileId) throws NotFoundException {
    *
    * @param profileId the id of the profile to save
    * @param profile the information of the profile
+   * @throws ProfileConflictException if trying to update the Native profile
```

Comment:
I think this type of check should happen earlier in the service. We don't need to start a transaction to know that this is not allowed.

Reply:
I also think this should result in a 405, not a 409. 405 means method not allowed, whereas 409 is some sort of conflict that might at some point get resolved.

Reply:
But for `saveProfile`, we should only throw this Exception when we find there is existing native profile in the dataset, which needs to happen in a transaction.

Reply:
hm ok i see.